torchvision/datapoints/_datapoint.py

```diff
@@ -132,6 +133,16 @@ def dtype(self) -> _dtype:  # type: ignore[override]
         with DisableTorchFunctionSubclass():
             return super().dtype
 
+    def __deepcopy__(self: D, memo: Dict[int, Any]) -> D:
+        # We need to detach first, since a plain `Tensor.clone` will be part of the computation graph, which does
+        # *not* happen for `deepcopy(Tensor)`. A side-effect from detaching is that the `Tensor.requires_grad`
+        # attribute is cleared, so we need to refill it before we return.
+        # Note: If the datapoint has extra metadata, there is no deepcopy of that happening here. These datapoint
+        # classes have to overwrite `__deepcopy__` themselves (possibly calling this method), unless the metadata
+        # values are immutable anyway, e.g. tuples or enums. In these cases deepcopy is a no-op and thus does not need
+        # to be handled explicitly. For example, `datapoints.BoundingBox` is relying on this behavior.
```

Comment:
I think this will be a bit overwhelming to process for readers. Maybe we can omit the part about user-defined datapoints, since we mostly care about our own datapoints here anyway? I think something like this could suffice:

> Note: we don't explicitly handle meta-data (like bbox format or spatial_size) because these are immutable types, already handled in clone().